src/wallet/rpcwallet.cpp

```diff
@@ -2500,6 +2511,107 @@ UniValue fundrawtransaction(const UniValue& params, bool fHelp)
+                            );
+
+    RPCTypeCheck(params, boost::assign::list_of(UniValue::VSTR));
+    uint256 hash;
+    hash.SetHex(params[0].get_str());
+
+    // get the transaction
+    if (!pwalletMain->mapWallet.count(hash))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid or non-wallet transaction id");
+    const CWalletTx& wtx = pwalletMain->mapWallet[hash];
+
+    // ownly bump the fee if the transation signals opt-in-RBF after BIP125
+    if (!SignalsOptInRBF(wtx))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction can't be replaced");
+
+    // get the old fee to allow oldfee+newfee
+    CAmount nDebit = wtx.GetDebit(ISMINE_SPENDABLE);
+    CAmount nOldFee = -(wtx.IsFromMe(ISMINE_SPENDABLE) ? wtx.GetValueOut() - nDebit : 0);
+    CAmount nFee = nOldFee; //set the old fee as the base fee
+
+    CMutableTransaction tx(wtx);
+    // remove scriptSigs, the signatures are invalid after mutating the transaction
+    for (std::vector<CTxIn>::iterator it(tx.vin.begin()); it != tx.vin.end(); ++it)
+    {
+        (*it).scriptSig = CScript();
+    }
+
+    // remove "old" change outputs
+    for (std::vector<CTxOut>::iterator it(tx.vout.begin()); it != tx.vout.end();)
+    {
+        if (pwalletMain->IsMine(*it) == ISMINE_SPENDABLE)
+            it = tx.vout.erase(it);
+        else
+            ++it;
+    }
+
+    string strFailReason;
+    int nChangePos = -1;
+
+    // re-fund the transaction, a new change output will be added
+    CReserveKey reservekey(pwalletMain);
+    if(!pwalletMain->FundTransaction(tx, reservekey, nFee, nChangePos, strFailReason, false))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, strFailReason);
+
+    // sign the new transaction
+    int nIn = 0;
+    CTransaction txNewConst(tx);
+    bool txSignSuccess = false;
+    for (std::vector<CTxIn>::iterator it(tx.vin.begin()); it != tx.vin.end(); ++it)
+    {
+        std::map<uint256, CWalletTx>::const_iterator mi = pwalletMain->mapWallet.find((*it).prevout.hash);
+        if (mi != pwalletMain->mapWallet.end() && (nIn < (int)(*mi).second.vout.size()))
+        {
+            // get the scriptPubKey of the prevout
+            const CScript& scriptPubKey = (*mi).second.vout[(*it).prevout.n].scriptPubKey;
+
+            CScript& scriptSigRes = tx.vin[nIn].scriptSig; //reference to the CMutableTx scriptSig
+            txSignSuccess = ProduceSignature(TransactionSignatureCreator(pwalletMain, &txNewConst, nIn, SIGHASH_ALL), scriptPubKey, scriptSigRes);
+
+            if (!txSignSuccess)
+                break;
+        }
+        nIn++;
+    }
+
+    if (!txSignSuccess)
+        throw JSONRPCError(RPC_WALLET_ERROR, "Can't sign transaction.");
+
+    // commit/broadcast the transaction
+    CWalletTx wtxBumped(pwalletMain,tx);
+    if (!pwalletMain->CommitTransaction(wtxBumped, reservekey))
+        throw JSONRPCError(RPC_WALLET_ERROR, "Error: The transaction was rejected! Did you already bump the fee for this transaction?");
```

Comment:
If you detect `(nFee - nOldFee) > minTxRelayFee` condition above, then it is not a question anymore. The message would be "Fee are already bumped to the actual fee rate".

The error on commit can be "Something went wrong when commiting the transaction". Generic and not descriptive, but it should not happen.